common/archival/client.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package archival
+
+import (
+	"context"
+	"errors"
+	"go.uber.org/cadence/.gen/go/shared"
+	"time"
+)
+
+// ArchiveRequest is request for Archive
+type ArchiveRequest struct {
+	DomainName   string
+	DomainID     string
+	WorkflowID   string
+	RunID        string
+	WorkflowType string
+	Status       string
+	CloseTime    time.Time
+	History      *shared.History
+}
+
+// Client is used to store and retrieve blobs
+type Client interface {
+	Archive(ctx context.Context, request *ArchiveRequest) error
```

Comment:
naming: how about we call this ArchiveWorkflow() or ArchiveWorkflowHistory() - just so that - in future if we archive something else (like domain metadata change history) - the APIs become less confusing ?